src/main/java/gregtech/api/util/GTJEIUtility.java

```diff
@@ -0,0 +1,144 @@
+package gregtech.api.util;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.world.DimensionType;
+import net.minecraft.world.biome.Biome;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.Loader;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static gregtech.api.GTValues.MODID_AR;
+
+/**
+ * Package for various helper methods used in Gregtech JEI integration
+ */
+public class GTJEIUtility {
+
+    private static final int FONT_HEIGHT = Minecraft.getMinecraft().fontRenderer.FONT_HEIGHT;
+
+    /**
+     * Creates a list of biomes whose weight, as described by the passed Function, differs from the passed original weight
+     * For use in the JEI Ore Spawn Page and JEI Fluid Spawn Page
+     *
+     * @param biomeFunction A Function describing the modified weights of biomes
+     * @param originalWeight The original weight of the biome
+     *
+     * @return A List containing all the modified biomes and their value they are modified by
+     */
+    public static List<String> createSpawnPageBiomeTooltip(Function<Biome, Integer> biomeFunction, int originalWeight) {
+
+        Iterator<Biome> biomeIterator = Biome.REGISTRY.iterator();
+        int biomeWeight;
+        Map<Biome, Integer> modifiedBiomeMap = new HashMap<>();
+        List<String> tooltip = new ArrayList<>();
+
+        //Tests biomes against all registered biomes to find which biomes have had their weights modified
+        while (biomeIterator.hasNext()) {
+
+            Biome biome = biomeIterator.next();
+
+            //Gives the Biome Weight
+            biomeWeight = biomeFunction.apply(biome);
+            //Check if the biomeWeight is modified
+            if (biomeWeight != originalWeight) {
+                modifiedBiomeMap.put(biome, originalWeight + biomeWeight);
+            }
+        }
+
+        if(!modifiedBiomeMap.isEmpty()) {
```

Comment:
This will be show even if there are no changed values because check for original weight is after this point
![image](https://user-images.githubusercontent.com/3093101/215237473-6b32f346-ac33-46e4-9be3-aa4e16dbbf21.png)